cp23/hpi-fun2/motivation/executables/generator.py

```diff
@@ -0,0 +1,232 @@
+        bosses.append(i)
+        test_case += f"{i} "
+    test_case += " ".join([str(x) for x in bosses]) + "\n"
+
+    filename = "switching_chain"
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Long chain of nodes. Each node has either weight 1 or -1. No path length is larger 1.")
+
+
+def negative_only_test():
+    test_case = ""
+
+    hierarchical_levels = 8
+    employees = 0
+    for level in range(0, hierarchical_levels):
+        employees += 3**level
+
+    threshold = 20
+    test_case += f"{employees} {threshold}\n"
+
+    # Each person has negative motivational level
+    levels = []
+    for i in range(0, employees):
+        levels.append(-i)
+    test_case += " ".join([str(x) for x in levels]) + "\n"
+
+    # I did not think about it at first, but actually there is
+    #       (employees - 3**(hierarchical_levels-1))*3 = employees - 1
+    # Each person has exactly 3 people under them.
+    bosses = []
+    for i in range(1, employees - 3**(hierarchical_levels-1)):
+        bosses.extend([i, i, i])
+    test_case += " ".join([str(x) for x in bosses]) + "\n"
+
+    filename = "negative_only"
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("In this tree, each nodes has three children. All weights are negative, which render no child unreachable.")
+
+
+def random_test_case(case_number, input_size, type):
+    test_case = ""
+
+    employees = input_size
+    threshold = random.randint(1, employees // 10)
+
+    test_case += f"{employees} {threshold}\n"
+
+    levels = []
+    for i in range(0, employees):
+        motivational_level = random.randint(-threshold, threshold)
+        levels.append(motivational_level)
+    test_case += " ".join([str(x) for x in levels]) + "\n"
+
+    bosses = [1]
+    for i in range(2, employees):
+        boss = random.randint(1, i-1)
+        bosses.append(boss)
+    test_case += " ".join([str(x) for x in bosses]) + "\n"
+
+    filename = f"random_test_case_{case_number}_{type}"
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text(f"Randomly generated tree: test case #{case_number}; n={employees}, threshold={threshold}, type={type}")
+
+
+# ------------------------
+# Small Test Cases
+# ------------------------
+
+# Relatively simple trees
+binary_tree_two_layers()
+unreachable_subtree_with_negatives()
+exponential_weights()
+single_node_bottleneck()
+single_parent_single_child()
+negative_root_node_weight()
+
+# ------------------------
+# Large Test Cases
+# ------------------------
+
+# Chain test cases: Tree, where each node has only one child
+large_increasing_chain()
+switching_chain()
+
+# Each node has three children and each motivational level of -1
+negative_only_test()
+
+# Random test cases
+# Limit Tests
+for i in range(1, 4):
+    random_test_case(i, INPUT_LIMIT, "large")
+  
+for i in range(4, 8):
+    random_test_case(i, random.randint(500, 10_000), "small")
```

Comment:
```suggestion
for i in range(1, 6):
    random_test_case(i, random.randint(500, 10_000), "small")
for i in range(6, 10):
    random_test_case(i, INPUT_LIMIT, "large")
```
This just ensures that smaller testcases are executed first which leads to faster judging in case of wrong-answers.